src/hotspot/share/asm/assembler.hpp

```diff
@@ -298,15 +310,22 @@ class AbstractAssembler : public ResourceObj  {
   // ensure buf contains all code (call this before using/copying the code)
   void flush();
 
-  void emit_int8(   uint8_t x1)                                     { code_section()->emit_int8(x1); }
+  void emit_int8(       int x1)                                     { code_section()->emit_int8(narrow_cast<uint8_t>(x1)); }
+
+  void emit_int16(       int x)                                     { code_section()->emit_int16(narrow_cast<uint16_t>(x)); }
 
-  void emit_int16(  uint16_t x)                                     { code_section()->emit_int16(x); }
-  void emit_int16(  uint8_t x1, uint8_t x2)                         { code_section()->emit_int16(x1, x2); }
+  void emit_int16(      int x1,     int x2)                         { code_section()->emit_int16(narrow_cast<uint8_t>(x1),
+                                                                                                 narrow_cast<uint8_t>(x2)); }
 
-  void emit_int24(  uint8_t x1, uint8_t x2, uint8_t x3)             { code_section()->emit_int24(x1, x2, x3); }
+  void emit_int24(      int x1,     int x2,     int x3)             { code_section()->emit_int24(narrow_cast<uint8_t>(x1),
+                                                                                                 narrow_cast<uint8_t>(x2),
+                                                                                                 narrow_cast<uint8_t>(x3)); }
 
   void emit_int32(  uint32_t x)                                     { code_section()->emit_int32(x); }
-  void emit_int32(  uint8_t x1, uint8_t x2, uint8_t x3, uint8_t x4) { code_section()->emit_int32(x1, x2, x3, x4); }
+  void emit_int32(      int x1,     int x2,     int x3,     int x4) { code_section()->emit_int32(narrow_cast<uint8_t>(x1),
+                                                                                                 narrow_cast<uint8_t>(x2),
+                                                                                                 narrow_cast<uint8_t>(x3),
+                                                                                                 narrow_cast<uint8_t>(x4)); }
```

Comment:
I'd rather add new alternative entry points for "int", so the existing callers using uint8_t don't need to perform unnecessary widening and narrowing.
Instead of narrow_cast<uint8_t>(x), how about (uint8_t)checked_cast<int8_t>(x)?

Reply:
How many callers are passing in negative values and actually need these convenience functions?

Reply:
> How many callers?

Actually quite a lot for the emit_int16/24/32 ones.  They pass an int imm8 value and have some (value | encode) parameter where encode is an int passed in.